caikit_nlp/modules/token_classification/filtered_span_classification.py

```diff
@@ -87,7 +87,7 @@ def __init__(
         error.type_check("<NLP79642537E>", ModuleBase, tokenizer=tokenizer)
         error.value_check(
             "<NLP42736791E>",
-            tokenizer.TASK_CLASS == TokenizationTask,
+            TokenizationTask in tokenizer._TASK_CLASSES,
```

Comment:
could we check `tasks`  now as opposed to checking the private attribute? ref. https://github.com/caikit/caikit/blob/main/caikit/core/modules/decorator.py#L157